### PR TITLE
[googlemaps] event MVCArray

### PR DESCRIPTION
**Pull request by demensky**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test YOUR_PACKAGE_NAME`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

If changing an existing definition:

- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://developers.google.com/maps/documentation/javascript/reference/event#MVCArray
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.
- [x] Include [tests for your changes](https://github.com/DefinitelyTyped/DefinitelyTyped#testing)
- [x] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`. If for reason the any rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]` and not for whole package so that the need for disabling can be reviewed.

### CODE REVIEW

File: types/googlemaps/reference/event.d.ts [hunk 1]
```diff
@@ -119,48 +119,143 @@ declare namespace google.maps {
...
+         * to {@link MVCArray.setAt setAt}() and the element that was previously in the array at that index.
+         * @see {@link https://developers.google.com/maps/documentation/javascript/reference/event#MVCArray.set_at Maps JavaScript API}
+         */
+        set_at: (this: C, index: number, previous: T) => void;
+    }
+
+    /**
+     * @see {@link https://developers.google.com/maps/documentation/javascript/reference/event#MVCArray Maps JavaScript API}
+     */
     class MVCArray<T> extends MVCObject {
```
Comment (demensky) [inline types/googlemaps/reference/event.d.ts:148]: @jpoehnelt. The prototype of the `MVCArray` object contains the `indexOf` method but is not included in [the documentation](https://developers.google.com/maps/documentation/javascript/reference/event#MVCArray). Is this method deprecated?

Console screen

**Comment by demensky** [comment]
@cgwrench @Silver-Connection @nertzy @xaolas @martincostello @svenkreiss @bolatovumar @gauthierm @captain-igloo @life777 @simonhaenisch @gshigeto @Bat-Orshikh @jpoehnelt — please [review this PR](https://github.com/DefinitelyTyped/DefinitelyTyped/pull/48551).

**Review by simonhaenisch — APPROVED** [review]
LGTM

**Comment by demensky** [comment]
Ready to merge

**Pull request merged**